test/running_examples.ts

```diff
@@ -2637,4 +2637,82 @@ ASSERT lines( lt_matches ) = 1.`;
     f(abap);
   });
 
+  it("Field offsets and lengths with structures, source", async () => {
+    const code = `
+  TYPES:
+    BEGIN OF ty_struct,
+      num TYPE i,
+    END OF ty_struct.
+  
+  DATA number TYPE i.
+  DATA struct TYPE ty_struct.
+  DATA test_string TYPE string.
+  
+  test_string = '0123456789'.
+  number = 3.
+  struct-num = 4.
+
+  WRITE / test_string+1(2).
+  WRITE / test_string+2(number).
+  WRITE / test_string+3(struct-num).
+
+  WRITE / test_string+number(2).
+  WRITE / test_string+number(number).
+  WRITE / test_string+number(struct-num).
+
+  WRITE / test_string+struct-num(2).
+  WRITE / test_string+struct-num(number).
+  WRITE / test_string+struct-num(struct-num).
+  
+  sy-index = 8. " This should probably not be allowed... :)
+  WRITE / test_string+sy-index(1).`;
+    const js = await run(code);
+    const f = new Function("abap", js);
+    f(abap);
+    expect(abap.console.get()).to.equal("12\n234\n3456\n34\n345\n3456\n45\n456\n4567\n8");
+  });
+
+  it("Field offsets and lengths with structures, target", async () => {
+    const code = `
+  TYPES:
+    BEGIN OF ty_struct,
+      num TYPE i,
+    END OF ty_struct.
+  
+  DATA number TYPE i.
+  DATA struct TYPE ty_struct.
+  DATA test_string TYPE string.
+  
+  test_string = '0123456789012'.
+  number = 3.
+  struct-num = 4.
+
+  test_string+1(2) = '##########'.
```

Comment:
`test_string` is a string, so abaplint should actually give a syntax error and not be able to execute this example

suggest changing test_string to a `TYPE c LENGTH 100`

Reply:
Oh, the things you don't know if you ever knew... 👍